syntax = "proto3";

service Agent {
  rpc Query(AgentRequest) returns (AgentResult);
}

message AgentRequest {
  string prompt = 1;
  int64 tag = 2;
}

message AgentResult {
  string new_prompt = 1;
  int64 tag = 2;
}
